### PR TITLE
[CIT-375] Add Azure Mapper document and some fixes

### PR DESCRIPTION
**Pull request by rina23q**
- Update the service names
- Correct all lower case mqtt to MQTT
- Correct broken links to test_connection doc

Signed-off-by: Rina Fujino <18257209+rina23q@users.noreply.github.com>

## Proposed changes

Add the descriptions for Azure IoT Hub Mapper and some fixes. 

## Types of changes

What types of changes does your code introduce to `thin-edge.io`?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactorings that doesn't explicitly fix a bug or add any new functionality)
- [x] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [ ] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: docs/src/architecture/mapper.md [hunk 1]
```diff
@@ -67,6 +67,52 @@ as both a fragment type and a fragment series in [Cumulocity's measurements](htt
...
 the message will be transferred to the topic `measurement/measurements/create` by [the MQTT bridge](../references/bridged-topics.md).
 
+## Azure IoT Hub mapper
+
+The Azure IoT Hub mapper takes messages formatted in the [Thin Edge JSON](thin-edge-json.md) as input.
+It validates if the incoming message is correctly formatted Thin Edge JSON, then outputs the message.
+The validated messages are published on the topic `az/messages/events/` from where they are forwarded to Azure IoT Hub.
+This mapper is launched by the `tedge connect az` command, and stopped by the `tedge disconnect az` command.
+
+The Azure IoT Hub Mapper works on this message in the following ways.
```
Comment (didier-wenzek) [inline docs/src/architecture/mapper.md:77]:
```suggestion
The Azure IoT Hub Mapper processes a message in the following ways.
```

**Review by didier-wenzek — COMMENTED** [review]
Looks good. I have just a small suggestion.

**Review by didier-wenzek — APPROVED** [review]
Approved

File: docs/src/architecture/mapper.md [hunk 2]
```diff
@@ -67,6 +67,52 @@ as both a fragment type and a fragment series in [Cumulocity's measurements](htt
...
+the output of the mapper is
+
+```json
+{
+	"temperature": 23,
+	"time": "2021-06-01T17:24:48.709803664+02:00"
+}
+```
+
+### Configure whether adding a timestamp or not
```
Comment (itsyitsy) [inline docs/src/architecture/mapper.md:102]:
```suggestion
### Configure whether to add a timestamp or not
```

**Pull request merged**